aspnetcore/blazor/hybrid/class-libraries.md

```diff
@@ -0,0 +1,514 @@
+        }
+    }
+}
+```
+
+The following example is for a .NET MAUI app in the solution that:
+
+* References the `SharedLibrary` project in its project file.
+* Registers the appropriate `CustomService` implmentation for each platform.
+
+In the app's project file (`.csproj`):
+
+```xml
+<ItemGroup>
+  <ProjectReference Include="..\SharedLibrary\SharedLibrary.csproj" />
+</ItemGroup>
+```
+
+In `MauiProgram` (`MauiProgram.cs`):
+
+```csharp
+using SharedLibrary.Interfaces;
+
+...
+
+#if WINDOWS
+    builder.Services.AddSingleton<SharedLibrary.ICustomService, 
+        SharedLibrary.Platforms.Windows.CustomService>();
+#elif ANDROID
+    builder.Services.AddSingleton<SharedLibrary.ICustomService, 
+        SharedLibrary.Platforms.Android.CustomService>();
+#elif MACCATALYST
+    builder.Services.AddSingleton<SharedLibrary.ICustomService, 
+        SharedLibrary.Platforms.MacCatalyst.CustomService>();
+#elif IOS
+    builder.Services.AddSingleton<SharedLibrary.ICustomService, 
+        SharedLibrary.Platforms.iOS.CustomService>();
+#endif
+```
+
+> [!NOTE]
+> Windows Forms Blazor projects register services in `Form1`'s contstructor (`Form1.cs`). WPF Blazor projects register services in `MainWindow`'s constructor (`MainWindow.xaml`).
+
+The following example for a Blazor Server or Blazor WebAssembly project:
+
+* References the `SharedLibrary` project in its project file.
+* Registers the appropriate `CustomService` implmentation for web-based clients.
+
+In the Blazor Server or Blazor WebAssembly app's project file (`.csproj`):
+
+```xml
+<ItemGroup>
+  <ProjectReference Include="..\SharedLibrary\SharedLibrary.csproj" />
+</ItemGroup>
+```
+
+In `Program.cs`, the project registers the custom service, `SharedLibrary.Web.CustomService`:
+
+```csharp
+builder.Services.AddSingleton<SharedLibrary.Interfaces.ICustomService, 
+    SharedLibrary.Web.CustomService>();
+```
+
+A component can inject the `CustomService` and call the `GetMessage` method. Add the following `TestCustomService` component to the shared `Pages` folder of the RCL.
+
+`Pages/TestCustomService.razor` in the RCL:
+
+```razor
+@page "/test-custom-service"
+@using SharedLibrary.Interfaces;
+@inject ICustomService CustomService
+
+<PageTitle>Test Service</PageTitle>
+
+<h3>Test Service</h3>
+
+<p>
+    <button class="btn btn-primary" @onclick="GetMessage">Get Message</button>
+</p>
+
+<p>
+    @message
+</p>
+
+@code {
+    private string message = string.Empty;
+
+    private void GetMessage()
+    {
+        message = CustomService.GetMessage();
+    }
+}
+```
+
+Add navigation for the `TestCustomService` component.
+
+In the `Shared/NavMenu.razor` file of the RCL:
+
+```razor
+<div class="nav-item px-3">
+    <NavLink class="nav-link" href="test-custom-service">
+        <span class="oi oi-list-rich" aria-hidden="true"></span> Test service
+    </NavLink>
+</div>
+```
+
+Regardless of where a C# class resides, a class can inject the `CustomService` and call `LogMessage`:
```

Comment:
btw --- This is a simple miss from prior updates. My first example used logging, so the method was `LogMessage`. I'll update this line to say `GetMessage` on my next commit (Tuesday morning).